File: frontend/src/dto/UserConditionsDTO.ts
import type { Direction } from "./Direction";
import type { UserPvConfig } from "./UserPvConfig";

export interface UserConditionsDTO {
    userPvConfig: UserPvConfig;   //
    montageAngle: number;         // 0-90
    montageDirection: Direction;  //
    montageShadeFactor: number;   // 0-1
}

File: backend/src/main/java/org/example/backend/dto/UserConditionsDTO.java
package org.example.backend.dto;

import jakarta.validation.constraints.*;
import org.example.backend.model.Direction;
import org.example.backend.model.UserPvConfig;

public record UserConditionsDTO(
        @NotNull
        UserPvConfig userPvConfig,

        @NotNull
        @Min(0)
        @Max(90)
        int montageAngle,

        @NotNull
        Direction montageDirection,

        @NotNull
        @DecimalMin("0.0")
        @DecimalMax("1.0")
        double montageShadeFactor
) {}
